nixos/modules/config/shells-environment.nix

```diff
@@ -107,14 +107,28 @@ in
     };
 
     environment.shellAliases = mkOption {
-      default = {};
-      example = { ll = "ls -l"; };
+      default = {
+        ls = "ls --color=tty";
+        ll = "ls -l";
+        l  = "ls -alh";
+      };
+      example = { gst = "git status"; };
       description = ''
         An attribute set that maps aliases (the top level attribute names in
         this option) to command strings or directly to build outputs. The
         aliases are added to all users' shells.
+
+        <note><para>
+        The defaults are there only as a suggestion and disappear when any alias is set.
+        To include the default aliases in your configuration use:
+        <code>
+        environment.shellAliases = {
+          gst = "git status";
+        } // options.environment.shellAliases.default;
+        </code>
+        </para></note>
       '';
-      type = types.attrs; # types.attrsOf types.stringOrPath;
+      type = types.attrs;
```

Comment:
This should use `types.attrsOf types.str` or something slightly less restrictive if necessary. That will make merges work like expected and I think also enables using `mkDefault` for attributes, like this:

```nix
{
    environment.shellAliases =
      { ls = mkDefault "ls --color=tty";
        ll = mkDefault "ls -l";
        l  = mkDefault "ls -alh";
      };
}
```